### PR TITLE
fix(tracer): downgrade default trace api from v05 to v04

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

Encoding traces in the v05 format has known issues for applications generating spans at a high frequency. Although this issue impacts 1/1000000 spans, these misformated payloads have the potential to degrade internal servoces.

## Motivation

Resolves an incident.

## Risk 

This change may introduce a small performance regression in some applications: https://github.com/DataDog/dd-trace-py/commit/1c12c0ab496de9fb168d82ec1fdfea18a328be61#diff-0b32c6cdae8bdd01786317bcabebf73dbb57e535adab74c94858fda05da411ef

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: releasenotes/notes/downgrade-encoding-to-v05-96458e1942295721.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+fixes:
+  - |
+    tracing: Encoding traces in the v05 format has known issues for applications generating spans at a frequency.
```
Comment (ZStriker19) [inline releasenotes/notes/downgrade-encoding-to-v05-96458e1942295721.yaml:4]: can we be more specific than "a frequency" ?

Comment (mabdinur) [reply releasenotes/notes/downgrade-encoding-to-v05-96458e1942295721.yaml:4]: We can reword this to say 1/1000000 spans

Comment (ZStriker19) [inline releasenotes/notes/downgrade-encoding-to-v05-96458e1942295721.yaml:4]:
```suggestion
    tracing: Encoding traces in the v05 format has a known issue for
    applications generating spans at a high frequency, causing approximately
    1/10000000 spans to be misencoded.
```

File: releasenotes/notes/downgrade-encoding-to-v05-96458e1942295721.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+    This change sets ``DD_TRACE_API_VERSION`` to ``v04`` by default. Approximately 1/10000000 spans are misencoded.
```
Comment (ZStriker19) [inline releasenotes/notes/downgrade-encoding-to-v05-96458e1942295721.yaml:5]:
```suggestion
    This change sets ``DD_TRACE_API_VERSION`` to ``v04`` by default. Approximately 1/10000000 spans are misencoded.
```
```suggestion
    This change sets ``DD_TRACE_API_VERSION`` to ``v04`` by default. 
```

**Review by ZStriker19 — DISMISSED** [review]

**Review by ZStriker19 — APPROVED** [review]
